addons/pos_self_order/models/product_product.py

```diff
@@ -101,3 +101,8 @@ def _send_availability_status(self):
                 config._notify('PRODUCT_CHANGED', {
                     'product.product': self.read(self._load_pos_self_data_fields(config.id), load=False)
                 })
+
+    def _find_record_check_access(self, access_token, field):
+        if field == "image_512":
+            return self.sudo()
```

Comment:
I know it was already like that before the PR but this seems generous, shouldn't it only be limited to products available for self ordering?

And in pos, only products available in pos?